File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/multiline/uncomment/direct/Java/Go/s580342216_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	solution := Solution{}
	solution.Run()
}

type Solution struct {
	scanner *bufio.Scanner
}

func (s *Solution) Run() {
	s.scanner = bufio.NewScanner(os.Stdin)
	for s.scanner.Scan() {
		s.runCase(s.scanner.Text())
	}
}

func (s *Solution) runCase(line string) {
	parts := strings.Fields(line)
	N, _ := strconv.Atoi(parts[0])
	K, _ := strconv.Atoi(parts[1])

	length := 0
	for N > 0 {
		N /= K
		length++
	}

	fmt.Println(length)
}

func strToIntArray(str string) []int {
	vals := strings.Fields(str)
	res := make([]int, len(vals))
	for i, val := range vals {
		res[i], _ = strconv.Atoi(val)
	}
	return res
}

type LCS struct {
	dp [][]int
}

func (l *LCS) lcs(A, B string) int {
	szA, szB := len(A), len(B)
	l.dp = make([][]int, szA+1)
	for i := range l.dp {
		l.dp[i] = make([]int, szB+1)
	}

	for i := 0; i <= szA; i++ {
		l.dp[i][0] = 0
	}
	for j := 0; j <= szB; j++ {
		l.dp[0][j] = 0
	}

	for i := 1; i <= szA; i++ {
		for j := 1; j <= szB; j++ {
			if A[i-1] == B[j-1] {
				l.dp[i][j] = l.dp[i-1][j-1] + 1
			} else {
				l.dp[i][j] = max(l.dp[i-1][j], l.dp[i][j-1])
			}
		}
	}

	return l.dp[szA][szB]
}

func (l *LCS) getLCS(X, Y string) string {
	m, n := len(X), len(Y)
	index := l.dp[m][n]
	temp := index

	lcs := make([]byte, index+1)
	lcs[index] = 0

	i, j := m, n
	for i > 0 && j > 0 {
		if X[i-1] == Y[j-1] {
			lcs[index-1] = X[i-1]
			i--
			j--
			index--
		} else if l.dp[i-1][j] > l.dp[i][j-1] {
			i--
		} else {
			j--
		}
	}

	return string(lcs[:temp+2])
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/uncomment/direct/Java/C++/s580342216_commented.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    Solution() {
        string line;
        while (getline(cin, line)) {
            run_case(line);
        }
    }
private:
    static vector<int> strToIntArray(const string& str) {
        istringstream iss(str);
        vector<int> res;
        int val;
        while (iss >> val) {
            res.push_back(val);
        }
        return res;
    }

    void run_case(const string& line) {
        istringstream iss(line);
        int N, K;
        if (!(iss >> N >> K)) return;
        int len = 0;
        while (N > 0) {
            N /= K;
            len++;
        }
        cout << len << '\n';
    }
};

class LCS {
public:
    int lcs(const string& A, const string& B) {
        int sz_a = A.length();
        int sz_b = B.length();
        dp.clear();
        dp.resize(sz_a + 1, vector<int>(sz_b + 1, 0));

        for (int i = 0; i <= sz_a; i++) {
            dp[i][0] = 0;
        }
        for (int j = 0; j <= sz_b; j++) {
            dp[0][j] = 0;
        }

        for (int i = 1; i <= sz_a; i++) {
            for (int j = 1; j <= sz_b; j++) {
                if (A[i - 1] == B[j - 1]) {
                    dp[i][j] = dp[i - 1][j - 1] + 1;
                } else {
                    dp[i][j] = max(dp[i - 1][j], dp[i][j - 1]);
                }
            }
        }
        return dp[sz_a][sz_b];
    }

    string getLCS(const string& X, const string& Y) {
        int m = X.length();
        int n = Y.length();
        int index = dp[m][n];
        string result;
        int i = m, j = n;
        while (i > 0 && j > 0) {
            if (X[i - 1] == Y[j - 1]) {
                result.push_back(X[i - 1]);
                i--;
                j--;
            } else if (dp[i - 1][j] > dp[i][j - 1]) {
                i--;
            } else {
                j--;
            }
        }
        reverse(result.begin(), result.end());
        return result;
    }
private:
    vector<vector<int>> dp;
};

int main() {
    Solution sol;
    return 0;
}
// End of Code
